src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -452,38 +463,52 @@ void VideoDecoder::addVideoStreamDecoder(
     throw std::invalid_argument("No valid stream found in input file.");
   }
   TORCH_CHECK(codec != nullptr);
+
+  StreamMetadata& streamMetadata = containerMetadata_.streams[streamNumber];
+  if (seekMode_ == SeekMode::approximate &&
+      !streamMetadata.averageFps.has_value()) {
+    throw std::runtime_error(
+        "Seek mode is approximate, but stream " + std::to_string(streamNumber) +
+        " does not have an average fps in its metadata.");
+  }
+
   StreamInfo& streamInfo = streams_[streamNumber];
   streamInfo.streamIndex = streamNumber;
   streamInfo.timeBase = formatContext_->streams[streamNumber]->time_base;
   streamInfo.stream = formatContext_->streams[streamNumber];
+
   if (streamInfo.stream->codecpar->codec_type != AVMEDIA_TYPE_VIDEO) {
     throw std::invalid_argument(
         "Stream with index " + std::to_string(streamNumber) +
         " is not a video stream.");
   }
+
   AVCodecContext* codecContext = avcodec_alloc_context3(codec);
-  codecContext->thread_count = options.ffmpegThreadCount.value_or(0);
   TORCH_CHECK(codecContext != nullptr);
```

Comment:
Note that this `TORCH_CHECK` was almost certainly optimized out. Because the check happens after a use, the compiler is allowed to infer that the pointer can never be null, so `codecContext != nullptr` can just become `true`. The check itself will then be removed as dead code. Null checks must happen before the first use.